pkg/auth/authenticator.go

```diff
@@ -69,3 +71,46 @@ func (ba *BuiltinAuthenticator) AuthenticateUser(ctx context.Context, username,
 func (ba *BuiltinAuthenticator) String() string {
 	return "built in authenticator"
 }
+
+// GetOrCreateUser searches for the user by username, and if not found, creates a new user with the given username and
+// external user identifier.  It returns the user.
+// This function is meant to be used by Authenticator implementations in the ChainAuthenticator.
+func GetOrCreateUser(ctx context.Context, authService Service, friendlyName, identifier, defaultUserGroup, source string) (*model.User, error) {
+	log := logging.FromContext(ctx).WithField("input_username", friendlyName)
+	dbUsername := friendlyName
```

Comment:
no no friendly name is friendly name. 
the dbUsername should be username (in another suggested  comment changed identifier -> username)